jetstream/experimenter.py

```diff
@@ -29,32 +81,101 @@ class Experiment:
+        control_slug = None
+
+        control_slugs = [variant.slug for variant in self.variants if variant.is_control]
+        if len(control_slugs) == 1:
+            control_slug = control_slugs[0]
+
+        return Experiment(
+            normandy_slug=self.normandy_slug,
+            slug=self.slug,
+            type=self.type,
+            status=self.status,
+            active=self.status == "Live",
+            start_date=self.start_date,
+            end_date=self.end_date,
+            proposed_enrollment=self.proposed_enrollment,
+            features=[],
+            branches=branches,
+            reference_branch=control_slug,
+        )
+
+
+@attr.s(auto_attribs=True, kw_only=True, slots=True, frozen=True)
+class NimbusExperiment:
+    """Represents a v4 experiment from Experimenter."""
+
+    slug: str  # Normandy slug
+    active: bool
+    features: List[Feature]
+    branches: List[Branch]
+    start_date: dt.datetime
+    end_date: Optional[dt.datetime]
+    proposed_enrollment: int
+    reference_branch: str
 
     @classmethod
-    def from_dict(cls, d) -> "Experiment":
+    def from_dict(cls, d) -> "NimbusExperiment":
         converter = cattr.Converter()
         converter.register_structure_hook(
-            dt.datetime, lambda num, _: cls._unix_millis_to_datetime(num),
+            dt.datetime, lambda num, _: _unix_millis_to_datetime(num),
         )
         return converter.structure(d, cls)
 
+    def to_experiment(self) -> "Experiment":
+        """Convert to Experiment."""
+        return Experiment(
+            normandy_slug=None,
+            slug=self.slug,
+            type="nimbus",
+            status=None,
+            active=self.active,
+            start_date=self.start_date,
+            end_date=self.end_date,
+            proposed_enrollment=self.proposed_enrollment,
+            features=self.features,
+            branches=self.branches,
+            reference_branch=self.reference_branch,
+        )
+
 
 @attr.s(auto_attribs=True)
 class ExperimentCollection:
     experiments: List[Experiment] = attr.Factory(list)
 
-    EXPERIMENTER_API_URL = "https://experimenter.services.mozilla.com/api/v1/experiments/"
+    EXPERIMENTER_API_URL_V1 = "https://experimenter.services.mozilla.com/api/v1/experiments/"
+
+    # for nimbus experiments
+    EXPERIMENTER_API_URL_V4 = "https://experimenter.services.mozilla.com/api/v4/experiments/"
 
     @classmethod
     def from_experimenter(cls, session: requests.Session = None) -> "ExperimentCollection":
         session = session or requests.Session()
-        experiments = session.get(cls.EXPERIMENTER_API_URL).json()
-        return cls([Experiment.from_dict(experiment) for experiment in experiments])
+        legacy_experiments_json = session.get(cls.EXPERIMENTER_API_URL_V1).json()
+        legacy_experiments = [
+            LegacyExperiment.from_dict(experiment).to_experiment()
+            for experiment in legacy_experiments_json
+        ]
```

Comment:
I think we should specifically exclude type="rapid" experiments from this list since they'll be in the v4 list.